internal/hardware/validator.go

```diff
@@ -150,11 +154,61 @@ func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infra
 		}
 	}
 
+	if disk.DriveType == models.DriveTypeISCSI {
+		err := isISCSINetworkingValid(disk, inventory)
+		if err != nil {
+			notEligibleReasons = append(notEligibleReasons, err.Error())
+		}
+	}
+
 	return notEligibleReasons, nil
 }
 
+// isISCSINetworkingValid checks if the iSCSI dish is not connected through the
+// default network interface. The default network interface is the interface
+// which is used by the default gateway.
+func isISCSINetworkingValid(disk *models.Disk, inventory *models.Inventory) error {
+	// get the IPv4 or the IPv6 of the interface connected to the iSCSI target
+	if disk.Iscsi == nil {
+		return fmt.Errorf("Host IP address is not available")
+
+	}
+	iSCSIHostIP, err := netip.ParseAddr(disk.Iscsi.HostIPAddress)
+	if err != nil {
+		return fmt.Errorf("Cannot parse iSCSI host IP %s: %w", disk.Iscsi.HostIPAddress, err)
+	}
+
+	defaultRoute := network.GetDefaultRouteByFamily(inventory.Routes, iSCSIHostIP.Is6())
+	if defaultRoute == nil {
+		return fmt.Errorf("Cannot find default route")
+	}
+
+	defaultInterface, ok := lo.Find(inventory.Interfaces, func(i *models.Interface) bool {
+		return i.Name == defaultRoute.Interface
+	})
+	if !ok {
+		return fmt.Errorf("Cannot find the network interface behind the default route")
+	}
+
+	// look if one of the IP assigned to the default interface interface
+	// corresponds to the IP used by host to connect on the iSCSI target
+	ips := defaultInterface.IPV4Addresses
+	if iSCSIHostIP.Is6() {
+		ips = defaultInterface.IPV6Addresses
+	}
+	_, ok = lo.Find(ips, func(ip string) bool {
+		prefix, err := netip.ParsePrefix(ip)
```

Comment:
Is there any reason why you decided to use netip.ParsePrefix and not net.ParseCIDR (it is just less known)

Reply:
I already use `netip` for `iSCSIHostIP` to check if the address is ipv4 or ipv6. So thought that it's would be consistent to use `netip` for all other operations in this function.